### PR TITLE
OMI-499: Open source omigami core

### PR DESCRIPTION
**Pull request by cereniyim**
This serves as a feature branch, do not push directly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,79 @@
...
+If a flow is using cache, these will show up in the logs. The tasks that use caching are:
+- `DownloadData`
+- `CreateChunks`
+- `CleanRawSpectra`
+- `CreateDocuments` [spec2vec specific]
+- `ProcessSpectrum` [ms2deepscore specific] - TODO
 
+Three values for dataset IDs are available for running locally:
+- `small`: a 100 spetra dataset
+- `small_500`: a 500 spectra dataset. I would've never guessed it.
```
Comment (cereniyim) [inline README.md:46]: a forgotten comment?

**Review by cereniyim — COMMENTED** [review]
Thank you Carlos, I left a comment

File: README.md [hunk 2]
```diff
@@ -107,24 +149,24 @@ To start a training flow, issue for instance:
...
-    --image "<image>"
-    --project-name=spec2vec 
-    --flow-name=spec2vec-first-flow-test
-    --dataset-id 10k
-    --ion-mode positive
+omigami spec2vec train \
+    --flow-name spec2vec-training \
+    --dataset-id small_500 \
+    --ion-mode positive \
+    --local
```
Comment (bernardolk) [inline README.md:169]: why you removed image arg here and left in the previous one?

File: omigami/spectra_matching/util.py [hunk 3]
```diff
@@ -95,3 +97,18 @@ def cosine_similarity_matrix(
     for i in range(vectors_2.shape[0]):
         vectors_2[i] = vectors_2[i] / norm_2[i]
     return np.dot(vectors_1, vectors_2.T)
+
+
+def run_local_training_flow(flow: Flow, project_name: str):
+    flow_run = flow.run()
+
+    register_task = flow.get_tasks("RegisterModel")[0]
+    run_id = flow_run.result[register_task].result
```
Comment (bernardolk) [inline omigami/spectra_matching/util.py:106]:
```suggestion
    mlflow_run_id = flow_run.result[register_task].result
```

Comment (bernardolk) [reply omigami/spectra_matching/util.py:106]: this is a suggestion : )

File: omigami/cli_options.py [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
-    click.option(
-        "--project-name",
-        "-p",
-        help="Name of the project. This is used as identification by Prefect",
-        required=True,
+        "--image",
+        "-i",
+        type=str,
+        required=False,
+        help="Image used to build the flow",
```
Comment (bernardolk) [inline omigami/cli_options.py:11]: can we add, here or on the readme, that this is required for running with prefect server?

**Review by bernardolk — COMMENTED** [review]
Thank you - added comments regarding small changes

**Pull request merged**